core/src/test/scala/org/apache/spark/api/python/SerDeUtilSuite.scala

```diff
@@ -32,5 +32,22 @@ class SerDeUtilSuite extends SparkFunSuite with SharedSparkContext {
     val pythonRdd = SerDeUtil.javaToPython(javaRdd)
     SerDeUtil.pythonToPairRDD(pythonRdd, false)
   }
+
+  test("Converting JavaRDD to PythonRDD with a fixed batch size of 1 (SPARK-48380)") {
+    val rddWith3Elements = sc.makeRDD(Seq[(Any, Any)]((1, 2), (3, 4), (5, 6)))
+    val javaRdd = rddWith3Elements.toJavaRDD()
+    val pythonRdd = SerDeUtil.javaToPython(javaRdd, 1)
+    val collected = pythonRdd.collect()
+    assert(collected.size() === 3)
+  }
+
+  test("Converting JavaRDD to PythonRDD with an auto batch size (SPARK-48380)") {
+    val rddWith3Elements = sc.makeRDD(Seq[(Any, Any)]((1, 2), (3, 4), (5, 6)))
+    val javaRdd = rddWith3Elements.toJavaRDD()
+    val pythonRdd = SerDeUtil.javaToPython(javaRdd)
+    val collected = pythonRdd.collect()
+    assert(collected.size() === 2)
```

Comment:
This is actually failing.  Need to find a way to verify that AutoBatchedPickler did batch rows together.

Reply:
Fixed by using a different test